<div class="alert alert-block alert-info">
Singapore Management University<br>
CS105 Statistical Thinking for Data Science, 2025/26 Term 2
</div>

# CS105 Group Project Submission (Part I)

-----
Provide your team details, including section, team number, team members, and the name of the dataset. 
Complete all of the following sections. For any part requiring code to derive your answers, please create a code cell immediately below your response and run the code.
To edit any markdown cell, double click the cell; after editing, execute the markdown cell to collapse it.
<br>
-----

## Declaration

<span style="color:red">By submitting this notebook, we declare that **no part of this submission is generated by any AI tool**. We understand that AI-generated submissions will be considered as plagiarism, and just like other plagirisum cases, disciplinary actions will be imposed.</span>

#### Section: 
#### Team: 
#### Members:
1. Chin Mun Yau
2. Hnit Oo Luck
3. Dong En Chua
4. 
5. 

#### Dataset: 

## Part I: Exploratory Data Analysis (EDA) [8% of final grade]

### 1. Overview of dataset [15% of Part I]

In [ ]:
import pandas as pd

wine_data = pd.read_csv("wine.csv")

**a.** Summarise the background of the dataset [limited to 50 words]

In [ ]:
wine_data

,ID,Fixed_Acidity,Volatile_Acidity,Citric_Acid,Residual_Sugar,Chlorides,Free_SO2,Total_SO2,Density,pH,Sulphates,Alcohol,Quality
0,1,7.0,0.27,0.36,High,0.045,45.0,170.0,>1.00,Acidic,0.45,8%,6.0
1,2,6.3,0.30,0.34,Low,0.049,14.0,132.0,0.99-1.00,Acidic,0.49,9%,6.0
2,3,8.1,0.28,0.40,Low,0.050,30.0,97.0,0.99-1.00,Acidic,0.44,10%,6.0
3,4,7.2,0.23,0.32,Moderate,0.058,47.0,186.0,0.99-1.00,Acidic,0.40,9%,6.0
4,5,7.2,0.23,0.32,Moderate,0.058,47.0,186.0,0.99-1.00,Acidic,0.40,9%,6.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...
4843,4844,6.2,0.21,0.29,Low,0.039,24.0,92.0,0.99-1.00,Acidic,0.50,11%,6.0
4844,4845,6.6,0.32,0.36,Moderate,0.047,57.0,168.0,0.99-1.00,Acidic,0.46,9%,5.0
4845,4846,6.5,0.24,0.19,Low,0.041,30.0,111.0,0.99-1.00,Highly Acidic,0.46,9%,6.0
4846,4847,5.5,0.29,0.30,Low,0.021,20.0,110.0,<0.99,Acidic,0.38,12%,7.0


<span style="color : red;"> Looking at the data from wine.csv, it suggests that it stores chemical and physical properties of different wines. </span>

**b.** State the size of the dataset

In [7]:
wine_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 4848 entries, 0 to 4847
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID                4848 non-null   int64  
 1   Fixed_Acidity     4848 non-null   float64
 2   Volatile_Acidity  4848 non-null   float64
 3   Citric_Acid       4848 non-null   float64
 4   Residual_Sugar    4591 non-null   str    
 5   Chlorides         4848 non-null   float64
 6   Free_SO2          4848 non-null   float64
 7   Total_SO2         4848 non-null   float64
 8   Density           4848 non-null   str    
 9   pH                4848 non-null   str    
 10  Sulphates         4848 non-null   float64
 11  Alcohol           4848 non-null   str    
 12  Quality           4590 non-null   float64
dtypes: float64(8), int64(1), str(4)
memory usage: 492.5 KB


<span style="color: red;">the dataset is 4848 rows x 13 columns large </span>

**c.** For each variable, describe what it represents and its data type (numerical or categorical)

<span style="color: red;"> 
there are 13 variables: <br>
1. ID - Unique ID of the wine (categorical) <br>
2. Fixed_Acidity - The amount of tartaric acid, measured in g/dm3 (numerical) <br>
3. Volatile_Acidity - Acetic acid, measured in g/dm3 (numerical) <br>
4. Citric_Acid - Citric acid, measured in g/dm3 (numerical) <br>
5. Residual_Sugar - Level of residual sugar content (categorical) <br>
6. Chlorides - Sodium Chloride levels, measured in mg/dm3 (numerical) <br>
7. Free_SO2 - Free SO2 ions measured in mg/dm3 (numerical) <br>
8. Total_SO2 - Total SO2 ions measured in mg/dm3 (numerical) <br>
9. Density - Density measured in g/dm3 (categorical) <br>
10. pH - pH value (categorical) <br>
11. Sulphates - Levels of potassium sulphate, measured in g/dm3 (numerical) <br>
12. Alcohol - Alcohol content (categorical) <br>
13. Quality - Quality score (categorical) <br>
</span>

### 2. Data pre-processing [35% of Part I]

**a.** For each variable, determine the percentage of missing data. For any column with missing data, describe how you resolve the issue. Clearly state any assumption you made.

In [13]:
# percentage of missing data
for i in range(13):
    wine_data.isna().sum()[i] / wine_data.sum()[i]

KeyError: 0

**Response.** 

**b.** For each variable, identify outliers (if any) and describe how you resolve the issue. Clearly state any assumption you made.

**Response.** 

**c.** For categorical variables, perform the necessary encoding.

**Response.** 

### 3.	Exploratory analysis and visualization [50% of Part I]

**a.** For each variable, provide relevant summary statistics

**Response.** 

**b.** For each variable, provide an appropriate visualisation depicting the distribution of its values, and summarize any key observation(s) you made.

**Response.** 

**c.** Perform bi-variate analysis on the variables. You do not need to present the analysis of every pair of variables; only focus on the pairs you believe are worth investigating and explain. For each pair, describe the relationship between the two variables. Use appropriate statistical methods and/or visualizations.

**Response.** 